helm/nessie/templates/deployment.yaml

```diff
@@ -129,6 +129,116 @@ spec:
                   name: {{ .Values.authentication.oidcClientSecret.name }}
                   key: {{ .Values.authentication.oidcClientSecret.key }}
             {{- end -}}
+            {{- if .Values.catalog.storage.s3.accessKeySecret.name }}
+            - name: NESSIE_CATALOG_SERVICE_S3_ACCESS_KEY_NAME
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.catalog.storage.s3.accessKeySecret.name }}
+                  key: {{ .Values.catalog.storage.s3.accessKeySecret.awsAccessKeyId }}
+            - name: NESSIE_CATALOG_SERVICE_S3_ACCESS_KEY_SECRET
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.catalog.storage.s3.accessKeySecret.name }}
+                  key: {{ .Values.catalog.storage.s3.accessKeySecret.awsSecretAccessKey }}
+            {{- end }}
+            {{- range $bucket := .Values.catalog.storage.s3.buckets -}}
+            {{- if $bucket.accessKeySecret }}
+            - name: NESSIE_CATALOG_SERVICE_S3_BUCKETS__{{- $bucket.name -}}__ACCESS_KEY_NAME
```

Comment:
I had to manually validate that this works. We can't put those names into upper-case letters because they are meant to be map keys, and a bucket name can be something like `bucket-1.prod-us`.

Note that such environment variable names are not Posix compliant (IEEE Std 1003.1-2001 says: uppercase letters, digits, and the '_' (underscore) sign only). But this doesn't seem to be a problem for the process running Nessie.

Reply:
At this point, I wonder if we shouldn't switch all our env vars to their "original" names. WDYT?

Reply:
Yes I've seen that. My fear was double-quotes in var names: we don't have any in the docker compose examples.
But I did a manual test and even that seems to work.
That said, OK I will change the new vars in this PR but let's maybe change the old ones in another PR.

Reply:
SGTM